components/sup/src/http_gateway.rs

```diff
@@ -301,32 +318,49 @@ fn json_response(data: String) -> HttpResponse {
 
 // Begin route handlers
 fn butterfly(req: &HttpRequest<AppState>) -> HttpResponse {
+    HTTP_GATEWAY_REQUESTS
+        .with_label_values(&["butterfly"])
+        .inc();
+    let timer = HTTP_GATEWAY_REQUEST_DURATION
+        .with_label_values(&["butterfly"])
+        .start_timer();
     let data = &req
         .state()
         .gateway_state
         .read()
         .expect("GatewayState lock is poisoned")
         .butterfly_data;
+    timer.observe_duration();
     json_response(data.to_string())
 }
 
 fn census(req: &HttpRequest<AppState>) -> HttpResponse {
+    HTTP_GATEWAY_REQUESTS.with_label_values(&["census"]).inc();
+    let timer = HTTP_GATEWAY_REQUEST_DURATION
+        .with_label_values(&["census"])
+        .start_timer();
```

Comment:
Yeah, it does, and that's a great observation. I've even used that feature before for other purposes, but I think it would fit really well for this purpose. I'm not sure how I overlooked that. I must've been on auto-pilot by this point.